controllers/nonadmin_controller.go

```diff
@@ -0,0 +1,182 @@
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+
+	oadpv1alpha1 "github.com/openshift/oadp-operator/api/v1alpha1"
+	"github.com/openshift/oadp-operator/pkg/common"
+)
+
+const (
+	nonAdminController           = "non-admin-controller"
+	nonAdminControllerDeployment = nonAdminController + "-deployment"
+	nonAdminControllerContainer  = nonAdminController + "-container"
+)
+
+var nonAdminControllerDeploymentLabel = map[string]string{
+	"component": nonAdminController,
+}
+
+func (r *DPAReconciler) ReconcileNonAdminController(log logr.Logger) (bool, error) {
+	// TODO https://github.com/openshift/oadp-operator/pull/1316
+	dpa := oadpv1alpha1.DataProtectionApplication{}
+	if err := r.Get(r.Context, r.NamespacedName, &dpa); err != nil {
+		return false, err
+	}
+
+	nonAdminDeployment := &appsv1.Deployment{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      nonAdminControllerDeployment,
+			Namespace: r.NamespacedName.Namespace,
+		},
+	}
+
+	// Delete (possible) previously deployment
+	if !r.checkNonAdminEnabled(&dpa) {
+		if err := r.Get(
+			context.Background(),
+			types.NamespacedName{
+				Name:      nonAdminDeployment.Name,
+				Namespace: nonAdminDeployment.Namespace,
+			},
+			nonAdminDeployment,
+		); err != nil {
+			if k8serror.IsNotFound(err) {
+				return true, nil
+			}
+			return false, err
+		}
+
+		deleteOptionPropagationForeground := metav1.DeletePropagationForeground
+		if err := r.Delete(
+			context.Background(),
+			nonAdminDeployment,
+			&client.DeleteOptions{PropagationPolicy: &deleteOptionPropagationForeground},
+		); err != nil {
+			r.EventRecorder.Event(
+				nonAdminDeployment,
+				corev1.EventTypeWarning,
+				"DeleteNonAdminDeploymentFailed",
+				fmt.Sprintf("Could not delete non admin controller deployment %s/%s: %s", nonAdminDeployment.Namespace, nonAdminDeployment.Name, err),
+			)
+			return false, err
+		}
+		r.EventRecorder.Event(
+			nonAdminDeployment,
+			corev1.EventTypeNormal,
+			"DeletedNonAdminDeploymentDeployment",
+			fmt.Sprintf("Non admin controller deployment %s/%s deleted", nonAdminDeployment.Namespace, nonAdminDeployment.Name),
+		)
+		return true, nil
+	}
+
+	operation, err := controllerutil.CreateOrUpdate(
+		r.Context,
+		r.Client,
+		nonAdminDeployment,
+		func() error {
+			// Setting Deployment selector if a new object is created, as it is immutable
+			if nonAdminDeployment.ObjectMeta.CreationTimestamp.IsZero() {
+				nonAdminDeployment.Spec.Selector = &metav1.LabelSelector{
+					MatchLabels: nonAdminControllerDeploymentLabel,
+				}
+			}
+
+			nonAdminImage := r.getNonAdminImage(&dpa)
+			err := r.buildNonAdminDeployment(nonAdminDeployment, nonAdminImage)
+			if err != nil {
+				return err
+			}
+
+			// Setting controller owner reference on the non admin controller deployment
+			return controllerutil.SetControllerReference(&dpa, nonAdminDeployment, r.Scheme)
+		},
+	)
+
+	if err != nil {
+		// TODO needed?
```

Comment:
copied this file from https://github.com/openshift/oadp-operator/blob/oadp-1.2/controllers/datamover.go

this part did not understand very well, it is a safe switch? but is it needed?

instead of checking `nonAdminDeployment.ObjectMeta.CreationTimestamp.IsZero()` to see if it is already created, could not we do a `client.Get`? would that make sure this is not needed, or there are corner cases this is needed?